src/pages/admin/submissions.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from "react";
+import { api } from "../../utils/api";
+import DefaultLoadingPage from "~/components/loading/loading";
+import {
+  TableRow,
+  TableCell,
+  TableContainer,
+  Table,
+  TableHead,
+  TableBody,
+  Paper,
+} from "@mui/material";
+
+const SubmissionPage = () => {
+  const {
+    data: collectionIds,
+    error: collectionError,
+    isLoading: collectionLoading,
+  } = api.collection.getAllCollectionId.useQuery();
+
+  const {
+    data: submissionsData,
+    error: submissionError,
+    isLoading: submissionLoading,
+  } = api.collection.getCollectionWithSubmissionsById.useQuery({
+    ids: collectionIds ? collectionIds.map((collection) => collection.id) : [],
+  });
+
+  if (collectionLoading || submissionLoading) return <DefaultLoadingPage />;
+  if (collectionError ?? submissionError)
+    return <div>Error: {submissionError?.message}</div>;
+
+  console.log(collectionIds);
+  console.log(submissionsData);
+
+  // Flatten the submissionsData array
+  const submissions = submissionsData.flatMap(
```

Comment:
You can try using ```useMemo``` instead of flatMap, as memorization can help avoid recomputing on component rerender. You can try
```
 const submissions = useMemo(() => submissionsData.flatMap(
    collection => collection.Submission
  ), [submissionsData]);
```